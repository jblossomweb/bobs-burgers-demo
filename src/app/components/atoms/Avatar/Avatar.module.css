.avatar {
  @apply relative
    rounded-full
    inline-block
    overflow-hidden;
}

.avatar :first-child {
  @apply flex
    justify-center
    items-center;
}
