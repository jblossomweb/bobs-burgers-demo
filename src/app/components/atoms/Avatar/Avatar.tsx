import React, { type CSSProperties } from 'react'
import colors from 'app/colors.json'
import style from './Avatar.module.css'

export interface Props {
  color?: CSSProperties['color']
  children: React.ReactNode
}

const Avatar: React.FC<Props> = ({
  children,
  color = colors['palette-light-blue'],
}) => (
  <div className={style.avatar} style={{ backgroundColor: color }}>
    {children}
  </div>
)

export default Avatar
