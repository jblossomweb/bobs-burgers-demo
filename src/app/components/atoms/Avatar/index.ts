/* eslint-disable no-restricted-exports */
export type { Props } from './Avatar'
export { default } from './Avatar'
