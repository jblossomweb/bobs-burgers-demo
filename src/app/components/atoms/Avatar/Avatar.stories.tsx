import React from 'react'
import type { ComponentStory, ComponentMeta } from '@storybook/react'

import Avatar, { type Props } from './Avatar'

export default {
  title: 'components/atoms/Avatar',
  component: Avatar,
  argTypes: {
    color: { control: 'color' },
    children: { control: { disable: true } },
  },
} as ComponentMeta<typeof Avatar>

const Template: ComponentStory<typeof Avatar> = (args: Props) => (
  <Avatar {...args} />
)

export const Image = Template.bind({})
Image.args = {
  // eslint-disable-next-line jsx-a11y/alt-text
  children: <img src="https://picsum.photos/80/80" />,
}

export const Letter = Template.bind({})
Letter.args = {
  children: <div style={{ width: '80px', height: '80px' }}>JB</div>,
}
