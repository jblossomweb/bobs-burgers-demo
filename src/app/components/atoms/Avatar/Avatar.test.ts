import storySnaps from 'test/utils/storySnaps'
import * as stories from './Avatar.stories'

describe('app/components/atoms/Avatar', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
