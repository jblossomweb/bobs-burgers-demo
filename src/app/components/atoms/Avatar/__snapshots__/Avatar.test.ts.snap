// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/atoms/Avatar stories Image matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="avatar"
    style="background-color: rgb(97, 218, 251);"
  >
    <img
      src="https://picsum.photos/80/80"
    />
  </div>
</DocumentFragment>
`;

exports[`app/components/atoms/Avatar stories Letter matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="avatar"
    style="background-color: rgb(97, 218, 251);"
  >
    <div
      style="width: 80px; height: 80px;"
    >
      JB
    </div>
  </div>
</DocumentFragment>
`;
